src/server/project.ts

```diff
@@ -102,6 +102,13 @@ namespace ts.server {
     export interface PluginModule {
         create(createInfo: PluginCreateInfo): LanguageService;
         getExternalFiles?(proj: Project): string[];
+        resolveModules?(createInfo: PluginCreateInfo): PluginResolveModules;

```

Comment:
the plugin already has access to the lshost, it can just override resolveModule directly. not sure i see the need for this new API.